pkg/v1/tkg/client/validate.go

```diff
@@ -1837,34 +1839,202 @@ func getDockerBridgeNetworkCidr() (string, error) {
 	return networkCidr, nil
 }
 
-// ConfigureAndValidateNetworkSeparationConfiguration validates the configuration of network separation
-func (c *TkgClient) ConfigureAndValidateNetworkSeparationConfiguration(clusterRole string) error {
-	if err := c.validateNetworkSeparation(constants.ConfigVariableAviManagementClusterServiceEngineGroup, clusterRole); err != nil {
+// ConfigureAndValidateAVIConfiguration validates the configuration inputs of Avi aka. NSX Advanced Load Balancer
+func (c *TkgClient) ConfigureAndValidateAviConfiguration() error {
+	aviEnable, _ := c.TKGConfigReaderWriter().Get(constants.ConfigVariableAviEnable)
+	// ignoring error because AVI_ENABLE is an optional configuration
+	if aviEnable == "" || aviEnable == "false" {
+		return nil
+	}
+
+	// if avi is enabled, then should verify following required fields
+	aviClient, err := c.ValidateAviControllerAccount()
```

Comment:
Is it possible to pass `aviClient` as an argument to the function `ConfigureAndValidateAviConfiguration` so that it increases testability?